### ISSUE TITLE
Support for Custom Objects

### ISSUE THREAD

**Issue by stobias123** [PaloAltoNetworks/terraform-provider-panos#157]
support for custom objects? 
Like URL categories.

**Comment by rfc1149** [comment]
+1 for URL categories. We use these quite heavily, and are something that gets updated relatively frequently for us. It would be great to move this configuration into terraform.

**Comment by shinmog** [comment]
@stobias123 @rfc1149 

Would a resource that manages the full object be ok, or do you need `entry` type objects where you can micro-manage individual items in the custom URL category?

**Comment by stobias123** [comment]
@shinmog  - IIRC the idea we were looking for was the ability to add/remove/modify a list of websites to a url category

**Comment by rfc1149** [comment]
Managing the full object would in my mind be basically equivalent to what we have now in panorama, where you can just add/remove entries on the list, and each entry is just the (wildcard) url with nothing else. 

Individual entry types could be a nice to have though. I've often thought while adding entries to a large URL Category list, such as "urls to exclude from TLS decryption" that it would be nice to be able to include a comment or note or link to a ticket reference or something like that so we could trace back in the future why something was added. This would be possible/easier if we had entry type objects in terraform.

TL;DR: Full object seems functionally equivalent to panorama, entry types could provide some nice benefits.

**Comment by mantisgb** [comment]
I would like Custom URL Categories with ability to create new categories as well as populate/remove items from the custom categories.

**Comment by shinmog** [comment]
Custom data objects added in v1.7.0.
